mvp-2003/Quikz
# Implement ESLint Commenting Workflow

## **Pull Request: Implement ESLint Commenting Workflow**

### **Description:**
This pull request introduces a GitHub Actions workflow that automatically runs ESLint checks on the backend code for every pull request. If any linting issues are found, the workflow will post a comment on the pull request, prompting contributors to run `npm run lint:fix` to resolve the issues. This enhancement aims to improve code quality and streamline the development process by ensuring that linting errors are addressed before merging.

### **Changes Made:**
- Created a GitHub Actions workflow that triggers on every pull request and push to all branches.
- Configured the workflow to run ESLint using the command `npx eslint .` in the `/src/server` directory.
- Added functionality to allow the workflow to pass even if linting issues are found, ensuring other checks can proceed.
- Implemented automated commenting on the pull request if any linting issues are detected, with guidance for contributors on how to fix them.

### **How to Test:**
1. Create a pull request with code that intentionally includes linting errors.
2. Check the pull request comments for the automated ESLint message.
3. Verify that the workflow runs successfully and posts comments as expected.

### **Related Issues:**
- Closes #46 

### **Additional Notes:**
- This workflow will help maintain consistent code quality across the project by encouraging contributors to fix linting issues before merging.
